src/models/UserProfileModel.ts

```diff
@@ -62,20 +69,18 @@ export class UserProfileModel extends Model {
      * @param element An element on which the model will be bound on.
      * @param options A set of options necessary for the component creation.
      */
-    constructor(element: HTMLElement, public options: IUserProfileModelOptions, public bindings: IComponentBindings) {
+    constructor(element: HTMLElement, public options: IUserProfileModelOptions) {
         super(element, UserProfileModel.ID, {});
-
-        const SEARCH_INTERFACE_ENDPOINT = this.bindings.searchInterface.options.endpoint;
-
-        this.options.restUri = this.options.restUri || this.parseEndpointUri(SEARCH_INTERFACE_ENDPOINT.options.restUri);
-        this.options.organization = this.options.organization || SEARCH_INTERFACE_ENDPOINT.options.queryStringArguments.organizationId;
+        Assert.isNotUndefined(this.options.restUri);
+        Assert.isNotUndefined(this.options.organizationId);
+        Assert.isNotUndefined(this.options.accessToken);
 
         this.getOrFetchCache = {};
 
         this.endpoint = new UserProfilingEndpoint({
             uri: this.options.restUri,
-            accessToken: SEARCH_INTERFACE_ENDPOINT.accessToken,
-            organization: this.options.organization
+            accessToken: this.options.accessToken || (this.options.searchEndpoint as SearchEndpoint).accessToken,
```

Comment:
Why the `as SearchEndpoint`, is accessToken not a prop of `ISearchEndpoint` ?

Reply:
Exactly only the class as accessToken

Reply:
Next question, this.options.searchEndpoint is always a defined object?

Reply:
Yes always